backend/de.metas.business/src/main/java-legacy/org/compiere/model/MPaymentTerm.java

```diff
@@ -16,311 +16,154 @@
  *****************************************************************************/
 package org.compiere.model;
 
-import static java.math.BigDecimal.ZERO;
-import static org.adempiere.model.InterfaceWrapperHelper.saveRecord;
-
-import java.math.BigDecimal;
-import java.sql.PreparedStatement;
-import java.sql.ResultSet;
-import java.util.ArrayList;
-import java.util.Properties;
-
-import javax.annotation.Nullable;
-
+import de.metas.invoice.InvoiceId;
+import de.metas.order.paymentschedule.InvoicePayScheduleService;
+import de.metas.payment.paymentterm.IPaymentTermRepository;
+import de.metas.payment.paymentterm.PaymentTermId;
+import de.metas.util.Services;
+import lombok.NonNull;
 import org.adempiere.ad.trx.api.ITrx;
-import org.adempiere.exceptions.AdempiereException;
+import org.compiere.SpringContextHolder;
 import org.compiere.util.DB;
-import org.compiere.util.Env;
 
-import de.metas.i18n.Msg;
-import lombok.NonNull;
+import javax.annotation.Nullable;
+import java.sql.ResultSet;
+import java.util.Properties;
 
+import static java.math.BigDecimal.ZERO;
 
 /**
- *	Payment Term Model
+ * Payment Term Model
  *
- *  @author Jorg Janke
- *  @version $Id: MPaymentTerm.java,v 1.3 2006/07/30 00:51:03 jjanke Exp $
+ * @author Jorg Janke
+ * @version $Id: MPaymentTerm.java,v 1.3 2006/07/30 00:51:03 jjanke Exp $
  */
 public class MPaymentTerm extends X_C_PaymentTerm
 {
-
+	@NonNull final InvoicePayScheduleService invoicePayScheduleService = SpringContextHolder.instance.getBean(InvoicePayScheduleService.class);
+	@NonNull final IPaymentTermRepository paymentTermRepository = Services.get(IPaymentTermRepository.class);
```

Comment:
pls avoid hooking services here.... they will be hooked each time we load one I_C_PaymetnTerm record